openedx/core/djangoapps/demographics/migrations/0002_clean_duplicate_entries.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from django.conf import settings
+from django.db import migrations, models
+
+log = logging.getLogger(__name__)
+
+
+def _clean_duplicate_entries(apps, schema_editor):
+    """
+    This method finds all the duplicate user entries in the UserDemographics model
+    and then removes all duplicate entries except for the most recently modified one.
+    """
+    demographics_model = apps.get_model('demographics', 'UserDemographics')
+    # Retrieve a list of all users that have more than one entry.
+    duplicate_users = (
+        demographics_model.objects.values(
+            'user'
+        ).annotate(models.Count('id')).values('user').order_by().filter(id__count__gt=1)
+    )
+    # Get a QuerySet of all the UserDemographics instances for the duplicates
+    # sorted by user and modified in descending order.
+    user_demographic_dupes = demographics_model.objects.filter(user__in=duplicate_users).order_by('user', '-modified')
+    curr_id = None
+
+    # Go through the QuerySet and only keep the most recent instance.
+    for demographic in user_demographic_dupes:
+        if demographic.user_id != curr_id:
+            log.info('UserDemographics {user} -- {modified}'.format(
+                user=demographic.user_id, modified=demographic.modified
+            ))
+            curr_id = demographic.user_id
+        else:
+            log.info('UserDemographics Duplicate User Delete {user} -- {modified}'.format(
+                user=demographic.user_id, modified=demographic.modified
+            ))
+            demographic.delete()
+
+
+class Migration(migrations.Migration):
+
+    initial = True
```

Comment:
This caught my eye. Do we need this `initial = True` here? I think I've only ever seen it on the very first migration for an app.